test/repeater-test.js

```diff
@@ -166,6 +166,36 @@ define(function(require){
 		});
 	});
 
+	test("should handle canceling search correctly", function () {
+		var count = -1;
+		var $repeater = $(this.$markup);
+		var $search = $repeater.find('.repeater-search');
+		$repeater.repeater({
+			dataSource: function(options, callback){
+				count++;
+				switch (count){
+					case 0:
+						equal(options.search, undefined, 'search value not passed to dataSource initially as expected');
+						callback({});
+						$search.find('input').val('something');
+						$search.trigger('searched.fu.repeater');
+						break;
```

Comment:
Cool. I started to write a while around it, but I wasn't sure if the unit tests had something like that built in. I'll check out that documentation tonight.